R/utils.R

```diff
@@ -129,31 +129,39 @@ render_dependencies <- function(dependencies, local = TRUE) {
     } else {
       "file"
     }
+    
+    # According to Dash convention, label react and react-dom as originating
+    # in dash_renderer package, even though all three are currently served
+    # up from the DashR package
+    if (dep$name %in% c("react", "react-dom")) {
+      dep$name <- "dash_renderer"
+    }
+
+    # The following lines inject _dash-component-suites into the src tags,
+    # as this is the current Dash convention. The dependency paths cannot
+    # be set solely at component library generation time, since hosted
+    # applications should have the app name injected as well.
+    #
+    # This is essentially analogous to this codeblock on the Python side:
+    # https://github.com/plotly/dash/blob/1249ffbd051bfb5fdbe439612cbec7fa8fff5ab5/dash/dash.py#L207
+    if (is.null(prefix)) {
+      dep[["script"]] <- paste0("/_dash-component-suites/",
+                                dep$name,
+                                "/",
+                                basename(dep[["script"]]))
+    } else {
+      dep[["script"]] <- paste0(sub("/", "", prefix), 
```

Comment:
I don't quite understand this `sub` - I believe we should be passing the prefix in directly here. if the prefix is `/my-app`, then that should be the start of this route, not `my-app`

Reply:
Hmm. This is what happens when I'm coding too late at night. I had thought `renderDependencies` was inserting a leading slash, but that's obviously incorrect (and didn't make much sense). This will all go away if we hand-code the script tags.

Without the prefix, a simple test shows

```
> htmltools:::renderDependencies(dependencies = dashR:::.dashR_js_metadata(), srcType = "file")`

<script src="lib/react%4015.4.2/dist/react.min.js"></script>
<script src="lib/react%4016.2.0/umd/react.production.min.js"></script>
<script src="lib/react-dom%4015.4.2/dist/react-dom.min.js"></script>
<script src="lib/react-dom%4016.2.0/umd/react-dom.production.min.js"></script>
<script src="lib/dash-renderer%400.18.0/dash_renderer/dash_renderer.min.js"></script>
```

But, in the codeblock you referenced, if `prefix` is `'/my-prefix/'` and we change it instead to 

```
dep[["script"]] <- paste0("/my-prefix/", "_dash-component-suites/", 
                           dep$name, "/", basename(dep[["script"]]))
```

The resulting tag (double slash at head) is

`<script src="//my-prefix/_dash-component-suites/dash_renderer/react.min.js"></script>`

...whereas if `prefix` is `'/'` we see

`<script src="//_dash-component-suites/dash_renderer/react.min.js"></script>`

Reply:
Ah OK, that's odd. Seems like another case in favor of us just rendering these script tag strings ourselves with `sprintf`

Reply:
fixed in 0569677c435c928878577db110b4036c9b0b12a8